sources/functions/bash_completion

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+#
+# Source the bash_completion installer
+# . /etc/swizzin/sources/functions/bash_completion
+#
+# Set the correct permissions for the completion file and then create a symlink to the bash completion file.
+if [ ! -e /etc/bash_completion.d/swizzin ]; then
+    ln -s '/etc/swizzin/sources/bash_completion/swizzin' '/etc/bash_completion.d/swizzin'
+    echo_info "Shell completions for the box command have been installed. They will be applied the next time you log into a bash shell"
+fi
+#
+# Create our apps and upgrade lists for the arrays.
+find "/etc/swizzin/scripts/install/" -type f -exec basename {} \; | sort | awk '{printf "%s ",$1 }' > /etc/swizzin/sources/bash_completion/completion.apps
+find "/etc/swizzin/scripts/upgrade/" -type f -exec basename {} \; | sort | awk '{printf "%s ",$1 }' > /etc/swizzin/sources/bash_completion/completion.upgrade
+#
+# Make sure the correct permissions are set
+chmod 755 '/etc/swizzin/sources/bash_completion'
+find "/etc/swizzin/sources/bash_completion/" -type f -exec chmod 644 {} \;
```

Comment:
Does this have to be done every single time? I think it's only necessary to do in the installer. Permissions should not change if you're `>`ing the files. The directory perms should stay the same too.

Reply:
I don't see why it would be a problem to make sure the permissions are reset and correct on an update.